### ISSUE TITLE
Frontend not building

### ISSUE THREAD

**Issue by GuillemPM** [origranot/url-shortener#188]
### Description

Since commit 93184f4efbc1953f56cadb55d2b068ce41b03fb4 frontend is not building

Gives the following error:

```
src/components/shortener-input/handleShortener.ts:83:13 - error TS2345: Argument of type '{ state: any; }' is not assignable to parameter of type 'Store'.
  Object literal may only specify known properties, and 'state' does not exist in type 'Store'.

83   copyUrl({ state });
               ~~~~~

src/routes/index.tsx:87:41 - error TS2345: Argument of type '{ state: Store; }' is not assignable to parameter of type 'Store'.
  Object literal may only specify known properties, and 'state' does not exist in type 'Store'.

87               onClick$={() => copyUrl({ state })}
                                           ~~~~~

src/routes/index.tsx:93:[43](https://github.com/origranot/url-shortener/actions/runs/3238344669/jobs/5306466998#step:4:44) - error TS23[45](https://github.com/origranot/url-shortener/actions/runs/3238344669/jobs/5306466998#step:4:46): Argument of type '{ state: Store; }' is not assignable to parameter of type 'Store'.
  Object literal may only specify known properties, and 'state' does not exist in type 'Store'.

93                 onClick$={() => copyUrl({ state })}
```

**Comment by cweave** [comment]
I'm having the issue of the index not compiling on change. I get two verrry different experiences. When vite launches, it opens `http://127.0.0.1:5173/`

![image](https://user-images.githubusercontent.com/16121690/195465421-563d4c9c-de06-46b4-bece-1d0481128e21.png)

and this hot reloads and all the good stuff.

But if I go to `http://localhost:3000/`, per the documentation, no hot reload, no updates.

![image](https://user-images.githubusercontent.com/16121690/195465512-3cc6cea5-0a86-4681-8cb3-fd9c17f5ef84.png)

Which one is the actual app? `http://127.0.0.1:5173/` matches demo, but can't hit the api with the server running (`404` error, but hits it on `localhost:3000`)

**Comment by GuillemPM** [comment]
Let me explain as it might be little bit complex.

We have 2 different apps in the same repo, Frontend (Qwik App) and Backend (NestJS), you can run each one separately as they have their own package.json, node_modules, etc. but to deploy the whole project we need to merge them into a single App, that's why we have a package.json in root folder. It means that if you launch the frontend separately you can't connect to the backend (as per now).

The root package.json is doing the following (`npm run build:all`):
Building the frontend to static files (`/dist` folder in frontend)
Building the backend (`/dist` folder in backend)
Copying the static files from the frontend to the backend (`/dist/public` folder in backend)

Then you can serve the single app using `npm run start:prod` (Always use npm run build:all before to have the most recent changes)

**Comment by GuillemPM** [comment]
I will close this issue as it's solved in PR #189 
If you have any doubt reach me opening a discussion here [Create discussion](https://github.com/origranot/url-shortener/discussions/new?category=q-a)
